packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -510,27 +492,83 @@ contract Colosseum is Initializable, Semver {
         require(sum == L2_ORACLE_SUBMISSION_INTERVAL, "Colosseum: invalid segments lengths");
     }
 
+    /**
+     * @notice Validates the output root proofs.
+     *
+     * @param _pos                Position of invalid section.
+     * @param _challenge          The challenge data.
+     * @param _srcOutputRootProof Proof of the source output root.
+     * @param _dstOutputRootProof Proof of the destination output root.
+     */
+    function _validateOutputRootProof(
+        uint256 _pos,
+        Types.Challenge storage _challenge,
+        Types.OutputRootProof calldata _srcOutputRootProof,
+        Types.OutputRootProof calldata _dstOutputRootProof
+    ) private view {
+        bytes32 srcOutputRoot = Hashing.hashOutputRootProof(_srcOutputRootProof);
+        bytes32 dstOutputRoot = Hashing.hashOutputRootProof(_dstOutputRootProof);
+
+        // If asserter timeout, the bisection of segments may not have ended.
+        // Therefore, segment validation only proceeds when bisection is not possible.
+        if (!_isAbleToBisect(_challenge)) {
+            require(
+                _challenge.segments[_pos] == srcOutputRoot,
+                "Colosseum: the source segment must be matched"
+            );
+            require(
+                _challenge.segments[_pos + 1] != dstOutputRoot,
+                "Colosseum: the destination segment must not be matched"
+            );
+        }
+
+        require(
+            _srcOutputRootProof.nextBlockHash == _dstOutputRootProof.blockHash,
+            "Colosseum: the block hash must be matched"
+        );
+    }
+
     /**
      * @notice Checks if the public input has been used before and validates if it is correct.
      *         Reverts if public input is used before or invalid.
```

Comment:
The comment does not match with the function.

Reply:
Good point. I modified comment for this